src/Planner/PlannerJoinTree.cpp

```diff
@@ -84,39 +84,47 @@ namespace
 {
 
 /// Check if current user has privileges to SELECT columns from table
-void checkAccessRights(const TableNode & table_node, const Names & column_names, const ContextPtr & query_context)
+/// Throws an exception if access to any column from `column_names` is not granted
+/// If `column_names` is empty, check access to any columns and retun names of accessible columns
+NameSet checkAccessRights(const TableNode & table_node, Names & column_names, const ContextPtr & query_context)
```

Comment:
What was the problem?

Reply:
When we execute query `select count() FROM table` we choose smallest column to read from `table`. But if we do not have an access to it we should choose another one, smallest from granted